File: pog-store/core/src/main/java/br/com/pogstore/eventStore/JsonEventSerializer.java
package br.com.pogstore.eventStore;

import java.nio.charset.StandardCharsets;

import br.com.pogstore.core.event.Event;
import br.com.pogstore.core.event.Snapshot;
import br.com.pogstore.core.storage.EventSerializer;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;

public class JsonEventSerializer implements EventSerializer {

	@Override
	public byte[] serializeEvent(Event event) {
		ObjectMapper mapper = new ObjectMapper();
		String json;

		try {
			json = mapper.writeValueAsString(event);
		} catch (JsonProcessingException e) {
			throw new RuntimeException(e.getMessage(), e);
		}

		byte[] bytes = json.getBytes(StandardCharsets.US_ASCII);

		return bytes;
	}

	@Override
	public Event deserializeEvent(byte[] data) {

		ObjectMapper mapper = new ObjectMapper();
		String decodedData = new String(data, StandardCharsets.US_ASCII);
		Event event;

		try {
			event = mapper.readValue(decodedData, Event.class);
		} catch (Exception e) {
			throw new RuntimeException(e.getMessage(), e);
		}

		return event;
	}

	@Override
	public Snapshot deserializeSnapshot(byte[] data) {

		ObjectMapper mapper = new ObjectMapper();
		String decodedData = new String(data, StandardCharsets.US_ASCII);
		Snapshot snapshot;

		try {
			snapshot = mapper.readValue(decodedData, Snapshot.class);
		} catch (Exception e) {
			throw new RuntimeException(e.getMessage(), e);
		}

		return snapshot;
	}

}

File: pog-store/cms/core/PogStore.Cms.EventStore/JsonEventSerializer.cs
using Newtonsoft.Json;
using PogStore.Cms.Core.Cqrs.Events;
using PogStore.Cms.Core.Cqrs.Storage;
using System.Text;

namespace PogStore.EventStore
{
	public class JsonEventSerializer : IEventSerializer
	{
		private JsonSerializerSettings _settings;

		public JsonEventSerializer()
		{
			_settings = new JsonSerializerSettings();
			_settings.TypeNameHandling = TypeNameHandling.All;
		}

		public byte[] SerializeEvent(IEvent @event)
		{
			var jsonEvent = JsonConvert.SerializeObject(@event, _settings);
			return Encoding.ASCII.GetBytes(jsonEvent);
		}

		public IEvent DeserializeEvent(byte[] data)
		{
			var jsonString = Encoding.ASCII.GetString(data);
			return (IEvent)JsonConvert.DeserializeObject(jsonString, _settings);
		}

		public Snapshot DeserializeSnapshot(byte[] data)
		{
			var jsonString = Encoding.ASCII.GetString(data);
			return (Snapshot)JsonConvert.DeserializeObject(jsonString, _settings);
		}
	}
}